### PR TITLE
Feature/mosaichunter

### PR DESCRIPTION
**Pull request by BasMonkey**
Added MosaiciHunter to WES Nextflow workflow

### CODE REVIEW

File: WES.config [hunk 1]
```diff
@@ -16,6 +16,11 @@ params {
     gatk_snp_cluster = "--clusterSize 3 --clusterWindowSize 35"
     gatk_indel_filter = "-filter 'QD < 2.0' -filterName INDEL_LowQualityDepth -filter 'FS > 200.0' -filterName INDEL_StrandBias -filter 'ReadPosRankSum < -20.0' -filterName INDEL_ReadPosRankSumLow"
 
+    mh_reference_file = "/hpc/diaggen/data/databases/ref_genomes/Homo_sapiens.GRCh37.GATK.illumina/Homo_sapiens.GRCh37.GATK.illumina.fasta"
+    mh_common_site_filter_bed_file = "/hpc/diaggen/users/bas/mosaictest/Dx_mosaic_project/mosaichunter/ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_100bpflank_flat.bed"
```
Comment (melferink) [inline WES.config:20]: Doesn't seem to be a production location.
Use Dx_tracks paths and refer to latest file including CREv4.
Include in Dx_tracks if necessary.

File: WES.config [hunk 2]
```diff
@@ -16,6 +16,11 @@ params {
...
+    mh_config_file_one = "/hpc/diaggen/users/bas/mosaictest/Dx_mosaic_project/mosaichunter/configurationfiles/exome_lh_ns_sb_parameters.conffile"
```
Comment (melferink) [inline WES.config:21]: see above. Not sure about location/repo though: I would suggest DxNextflowWES/assests, but let's discuss with the others first

**Review by melferink — DISMISSED** [review]
Few small comments. 
Main will be the correct production locations

File: WES.config [hunk 3]
```diff
@@ -16,6 +16,11 @@ params {
...
+    mh_config_file_two = "/hpc/diaggen/users/bas/mosaictest/Dx_mosaic_project/mosaichunter/configurationfiles/exome_lh_ns_sb.conffile"
```
[the remark this answers is not in the document]
Comment (BasMonkey) [reply WES.config]: Moved to DxTracks repo

File: WES.config (hunk 1 above)
Comment (BasMonkey) [reply WES.config]: Moved / changed to DxTracks

File: WES.config [hunk 4]
```diff
@@ -16,6 +16,16 @@ params {
...
+    mh_reference_file = "$params.genome"
+    mh_common_site_filter_bed_file = "$params.dxtracks_path/Tracks/ENSEMBL_UCSC_merged_collapsed_sorted_v3_CREv2_SSv7_CREv4_add_100bpflank_flat.bed"
+    mh_config_file_one = '$workflow.baseDir/CustomModules/MosaicHunter/1.0.0/exome_lh_ns_sb_parameters.conffile"
+    mh_config_file_two = "$workflow.baseDir/CustomModules/MosaicHunter/1.0.0/mosaichunter/configurationfiles/exome_lh_ns_sb.conffile"
+    mh_gender_ratio_x_threshold_male = 5.5
+    mh_gender_ratio_x_threshold_female = 7.5
+    mh_gender_locus_x = "X:2699520-154931044"
+    mh_gender_mapping_qual = 20
+
+
```
Comment (rernst) [inline WES.config:28]: 1 empty line is enough :D

File: WES.config [hunk 5]
```diff
@@ -271,6 +281,30 @@ process {
...
+        time = { (7.h * task.attempt) }
+        clusterOptions = "$params.cluster_options --gres=tmpspace:15G"
+    }
+
+    withLabel: MosaicHunterMosaicVariantCalculation {
+        cpus = 2
+        memory = { 20.GB * task.attempt }
+        time = { (7.h * task.attempt) }
+        clusterOptions = "$params.cluster_options --gres=tmpspace:15G"
+    }
```
Comment (rernst) [inline WES.config:306]: Add publish dir?

File: WES.nf [hunk 6]
```diff
@@ -200,6 +201,35 @@ workflow {
...
+    )
+
+    // Execute MH step one, Data Quality Correction
+    MosaicHunterQualityCorrection(
+        Sambamba_Merge.out.groupTuple(),
+        "$params.mh_reference_file",
+        "$params.mh_common_site_filter_bed_file",
+        "$params.mh_config_file_one",
+        MosaicHunterGetGender.out
+    )
```
Comment (rernst) [inline WES.nf:221]: Duplicate code? I think you are missing `MosaicHunterGetGender`?

**Review by rernst — CHANGES_REQUESTED** [review]
I think you are missing some components?

File: WES.config (hunk 5 above)
Comment (BasMonkey) [reply WES.config:305]: `publishDir` is defined in the process, in order to use the `saveAs` parameter to add the `sample_id` in the saved filename. This is not be possible when `publishDir` is placed in the `WES.config` file.

File: WES.nf (hunk 6 above)
Comment (BasMonkey) [reply WES.nf]: Whoops, fixed this!

File: WES.config [hunk 7]
```diff
@@ -16,6 +16,16 @@ params {
     gatk_snp_cluster = "--clusterSize 3 --clusterWindowSize 35"
     gatk_indel_filter = "-filter 'QD < 2.0' -filterName INDEL_LowQualityDepth -filter 'FS > 200.0' -filterName INDEL_StrandBias -filter 'ReadPosRankSum < -20.0' -filterName INDEL_ReadPosRankSumLow"
 
+    mh_reference_file = "$params.genome"
```
[the remark this answers is not in the document]
Comment (BasMonkey) [reply WES.config:19]: I created this additional variable to keep the MosaicHunter module more flexible. Now a different reference_file can easily be selected specifically for MosaicHunter without having to adjust the process. But if it's better to just use `$params.genome`, I'll change it :)

File: WES.nf [hunk 8]
```diff
@@ -202,32 +208,25 @@ workflow {
...
 
     // Execute MH step one, Data Quality Correction
     MosaicHunterQualityCorrection(
-        Sambamba_Merge.out.groupTuple(),
+        Sambamba_Merge.out.join(MosaicHunterGetGender.out).groupTuple(),
         "$params.mh_reference_file",
         "$params.mh_common_site_filter_bed_file",
-        "$params.mh_config_file_one",
-        MosaicHunterGetGender.out
+        "${workflow.projectDir}$params.mh_config_file_one"
```
Comment (rernst) [inline WES.nf:221]: Ik zou hier voor de zekerheid altijd een `/` tussen zetten.

File: WES.nf [hunk 9]
```diff
@@ -202,32 +208,25 @@ workflow {
...
     // Execute MH step two, Mosaic Variant Calculation
     MosaicHunterMosaicVariantCalculation(
-        Sambamba_Merge.out.groupTuple(),
+        Sambamba_Merge.out.join(MosaicHunterGetGender.out).join(MosaicHunterQualityCorrection.out),
         "$params.mh_reference_file",
         "$params.mh_common_site_filter_bed_file",
-        "$params.mh_config_file_two",
-        MosaicHunterQualityCorrection.out,
-        MosaicHunterGetGender.out
+        "${workflow.projectDir}$params.mh_config_file_two"
```
Comment (rernst) [inline WES.nf:229]: Zie hierboven.

**Review by rernst — CHANGES_REQUESTED** [review]
Nog een klein dingetje.

File: WES.nf (hunk 8 above)
Comment (BasMonkey) [reply WES.nf]: Toegevoegd :)

File: WES.nf (hunk 9 above)
Comment (BasMonkey) [reply WES.nf]: Toegevoegd :)

**Review by rernst — APPROVED** [review]

**Review by ellendejong — APPROVED** [review]

**Pull request merged**